pkg/fake/utils.go

```diff
@@ -155,6 +155,15 @@ func Filter(filters []ec2types.Filter, id, name, owner, state string, tags []ec2
 	})
 }
 
+func FilterCapacityReservation(filters []ec2types.Filter, id, name, owner, state, instanceMatchCriteria string, tags []ec2types.Tag) bool {
+	return lo.EveryBy(filters, func(filter ec2types.Filter) bool {
+		if aws.ToString(filter.Name) == "instance-match-criteria" {
+			return lo.Contains(filter.Values, instanceMatchCriteria)
+		}
+		return Filter([]ec2types.Filter{filter}, id, name, owner, state, tags)
+	})
+}
```

Comment:
nit: since `Filter` is already iterating over all of the filters, I'd just do this:
```suggestion
	return lo.EveryBy(filters, func(filter ec2types.Filter) bool {
		if aws.ToString(filter.Name) == "instance-match-criteria" {
			return lo.Contains(filter.Values, instanceMatchCriteria)
		}
		return true
	}) && Filter(filters, id, name, owner, state, tags)
}
```